server/src/js/dataSource.js

```diff
@@ -20,6 +20,16 @@ fluid.defaults("sjrk.storyTelling.server.dataSource.couch.base", {
         getURL: {
             funcName: "sjrk.storyTelling.server.dataSource.couch.base.getURL",
             args: ["{that}.options.host", "{that}.options.path"]
+        },
+        logDataSourceError: {
+            funcName: "fluid.log",
+            args: [fluid.logLevel.WARN, "Datasource error: ", "{arguments}.0"]
+        }
+    },
+    listeners: {
+        "onError.logDataSourceError": {
+            func: "{that}.logDataSourceError",
```

Comment:
You can get rid of the `logDataSourceError` invoker by replacing line 31, 32 with line 25, 26.

Reply:
I agree that this will make the code shorter, but I'm unsure since that would make this particular task more closely coupled to the onError event itself. Then again, it is "just" a call to fluid.log. I can't think of a case at the moment where we'd want to call the invoker in a way that isn't associated with the onError event on this component.